src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzTabularAclStep.java

```diff
@@ -53,6 +66,16 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
     emails.add(policies.get(IamRole.STEWARD));
     emails.add(policies.get(IamRole.READER));
 
+    Boolean inheritEnabled =
+        context
+            .getInputParameters()
+            .get(SnapshotWorkingMapKeys.SNAPSHOT_INHERIT_STEWARD_ENABLED, Boolean.class);
+    if (inheritEnabled != null && inheritEnabled) {
+      var datasetPolicyMap =
+          iamService.retrievePolicyEmails(userReq, IamResourceType.DATASET, sourceDataset.getId());
+      emails.add(datasetPolicyMap.get(IamRole.CUSTODIAN));
+    }
+
```

Comment:
Should we pull this into a common method somewhere?

Reply:
This code is temporary until https://broadworkbench.atlassian.net/browse/DT-1197 is done. That will replace this code, which reads from the flight input map, to read from a field in `sourceDataset`. I'm expecting it would something like
```
    if (sourceDataset.inheritSteward()) {
      var datasetPolicyMap =
          iamService.retrievePolicyEmails(userReq, IamResourceType.DATASET, sourceDataset.getId());
      emails.add(datasetPolicyMap.get(IamRole.CUSTODIAN));
    }
```

So I think a lot of it will go away. If there's still a need to refactor common code it can be done then.